portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -3397,10 +3413,23 @@ private DLFileEntry _updateFileEntry(
 
 			Date date = new Date();
 
+			String actualExtension = null;
+
+			if (inputStream instanceof ByteArrayFileInputStream) {
+				ByteArrayFileInputStream inputByteArrayFileInputStream =
+					(ByteArrayFileInputStream)inputStream;
+
+				File uploadedFile = inputByteArrayFileInputStream.getFile();
+
+				String actualName = uploadedFile.getName();
```

Comment:
Same as before; this particular one can be inlined.

Reply:
Changed